app/controlplane/pkg/data/ent/schema/membership.go

```diff
@@ -49,18 +50,28 @@ func (Membership) Fields() []ent.Field {
 			}),
 		// rbac role in the organization
 		field.Enum("role").GoType(authz.Role("")),
+
+		// polymorphic membership for RBAC
+		field.Enum("membership_type").GoType(biz.MembershipType("")).Optional(),
+		field.UUID("member_id", uuid.UUID{}).Optional(),
+
+		field.Enum("resource_type").GoType(biz.ResourceType("")).Optional(),
+		field.UUID("resource_id", uuid.UUID{}).Optional(),
 	}
 }
 
 func (Membership) Edges() []ent.Edge {
 	return []ent.Edge{
+		// Deprecated: use polymorphic membership instead
```

Comment:
how difficult do you think it would be to kill those columns? 

I am ok if we keep it in multiple steps though

Reply:
I think I can create an additional migration for it.